File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-16.cpp
//Write a program that reads a number and prints all of its binary digits : Print the
//remainder  number % 2, then replace the number with  number / 2.Keep going until the
//number is 0. For example, if the user provides the input 13, the output should be
//1
//0
//1
//1

#include<iostream>

using namespace std;

int main()
{
    cout << "Enter a number: ";
    int number;
    cin >> number;

    while(number !=  0)
    {
        cout << number % 2;
        number = number / 2;
    }
    return 0;
}

File: Python For Everyone, Horstmann/Chapter4/P4.16.py
# Factoring of integers. Write a program that asks the user for an integer and then
# prints out all its factors. For example, when the user enters 150, the program should
# print
#     2
#     3
#     5
#     5

inputN = int(input("Enter an integer: "))

i = 2

while i <= inputN:
    if inputN % i == 0:
        print(i)
        inputN /= i

    else:
        i += 1